src/node.cc

```diff
@@ -3989,6 +3992,21 @@ void FreeEnvironment(Environment* env) {
   delete env;
 }
 
+#if defined(__POSIX__) && defined(NODE_SHARED_MODE)
+void StopHelper(uv_async_t* handle) {
+  FreeEnvironment(static_cast<Environment*>(handle->data));
```

Comment:
We should not call `FreeEnvironment()` while we’re inside `uv_run()` and access the environment afterwards (which is what currently has to happen if you’re using public APIs, right?)

Reply:
yes, I too am concerned about the Environment pointer being stale in the middle of execution. So any pointers? is `RunCleanup` alone is sufficient here?

Reply:
I think it might be nice to define the semantics first, and then look at the implementation?

`uv_stop()` might be a good starting point here, but then again, that’s already something that you can do as an embedder without this patch